ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -23,19 +23,44 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
    * Load a target
    * @param {object} params
    * @param {string} params.target_id
-   * @param {bool} params.isConnecting
    * @return {TargetModel}
    */
-  async model({ target_id, isConnecting }) {
-    const target = await this.store.findRecord('target', target_id, {
+  model({ target_id }) {
+    return this.store.findRecord('target', target_id, {
       reload: true,
     });
+  }
 
-    if (isConnecting) {
-      await this.preConnect(target);
-    }
+  async afterModel(model, transition) {
+    const { isConnecting } = transition.to.queryParams;
+
+    if (isConnecting && model.address) {
+      await this.connect(model);
+    } else {
+      const hostSets = await Promise.all(
+        model.host_sources.map(({ host_source_id }) =>
+          this.store.findRecord('host-set', host_source_id)
+        )
+      );
+
+      // Extract host ids from all host sets
+      const hostIds = hostSets.map(({ host_ids }) => host_ids);
+
+      // Load unique hosts
+      const uniqueHostIds = new Set(hostIds.flat());
```

Comment:
I am not understanding why we use `new Set` here. As per my understanding `.flat()` will return a new array, but `new Set` will return an object.

Later on (below) you use `uniqueHostIds` with `array.map()` which needs and array. Maybe we can avoid uniqueHostsIds types transformations and just operate with array?

Reply:
Yeah I would like to do some testing with this. I simply added back the host fetching logic we had for the `targets/target-id/hosts` route we had in the past. The `Set()` is to remove any duplicate hostIds. However, I am not sure if we can have duplicates so I would need to confirm if that is even a concern. I would like to avoid transforming the data back and forth if possible. I will look into this to see if this was done for performance reasons.

Reply:
@cameronperera gotcha. I think we could do some pair programming next week? I am missing a bit of context and a quick pair session will help to understand better the purpose of this 😄